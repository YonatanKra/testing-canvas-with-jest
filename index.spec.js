const { draw } = require('./index');

describe('draw', () => {
    let canvas, ctx;
    beforeEach(function() {
        canvas = document.createElement('canvas');
        ctx = canvas.getContext('2d');
    });

    it(`should return the shape's path`, function() {
        const shapePath = draw(ctx);
        expect(shapePath instanceof Path2D).toBeTruthy();
    });

    it(`should draw a house on the canvas using the main ctx`, function() {
        draw(ctx);
        const events = ctx.__getEvents();

        expect(events).toMatchSnapshot();
    });

    });
});
