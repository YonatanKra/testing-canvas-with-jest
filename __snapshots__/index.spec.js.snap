// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draw should draw a house on the canvas using the main ctx 1`] = `
Array [
  Object {
    "props": Object {
      "value": "#808080",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  Object {
    "props": Object {
      "value": "#ffffff",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  Object {
    "props": Object {
      "value": 2,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  Object {
    "props": Object {
      "value": "#ffff00",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  Object {
    "props": Object {
      "path": Array [],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  Object {
    "props": Object {
      "path": Array [],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;
